src/main/java/org/broadinstitute/hellbender/transformers/BaseQualityClipReadTransformer.java

```diff
@@ -0,0 +1,110 @@
+package org.broadinstitute.hellbender.transformers;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.util.StringUtil;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.tools.ClipReads;
+import org.broadinstitute.hellbender.utils.clipping.ClippingOp;
+import org.broadinstitute.hellbender.utils.clipping.ClippingRepresentation;
+import org.broadinstitute.hellbender.utils.clipping.ReadClipper;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import static org.broadinstitute.hellbender.utils.read.ReadUtils.*;
+import static org.broadinstitute.hellbender.utils.read.ReadUtils.setDeletionBaseQualities;
+
+/**
+ * Created by markw on 10/21/16.
+ */
+
+public class BaseQualityClipReadTransformer implements ReadTransformer {
+    public static final long serialVersionUID = 1L;
+
+
+    /**
+     * The quality score based read clipper will be applied to the reads using this
+     * quality score threshold.
+     */
+    @Argument(fullName = "qTrimmingThreshold", shortName = "QT", doc = "Quality score trimmer threshold", optional = true)
+    private int qTrimmingThreshold = 15;
+
+    /**
+     * Clip bases from the read from
+     * <p/>
+     * argmax_x{ \sum{i = x + 1}^l (qTrimmingThreshold - qual)
+     * <p/>
+     * to the end of the read.  This is blatantly stolen from BWA.
+     * <p/>
+     * Walk through the read from the end (in machine cycle order) to the beginning, calculating the
+     * running sum of qTrimmingThreshold - qual.  While we do this, we track the maximum value of this
+     * sum where the delta > 0.  After the loop, clipPoint is either -1 (don't do anything) or the
+     * clipping index in the read (from the end).
+     *
+     */
+    @Override
+    public GATKRead apply(GATKRead read) {
+        GATKRead readClippedRightEnd = clipReadRightEnd(read);
+        return clipReadLeftEnd(readClippedRightEnd);
```

Comment:
The reverse strand flag will never be set in an unaligned BAM, so the read will be right clipped, which is correct because the read will be presented in machine cycle order.  (Reads only get reverse complemented by the aligner when it discovers that they map to the non-canonical strand of the reference.)
Actually, you may need to consider this fact when you're re-writing the reads as unmapped reads:  you might want to RC any of the output reads that have been RC'd.